core/ledger/pvtdatastorage/helper.go

```diff
@@ -181,6 +189,41 @@ func prepareHashedIndexEntries(dataEntires []*dataEntry) ([]*hashedIndexEntry, e
 	return hashedIndexEntries, nil
 }
 
+func preparePurgerMarkerEntries(blkNum uint64, purgeMarkers []*PurgeMarker) ([]*purgeMarkerEntry, []*purgeMarkerCollEntry) {
+	purgeMarkersEntries := []*purgeMarkerEntry{}
+	purgeMarkersCollEntries := []*purgeMarkerCollEntry{}
+
+	for _, m := range purgeMarkers {
+		purgeMarkersEntries = append(purgeMarkersEntries,
+			&purgeMarkerEntry{
+				key: &purgeMarkerKey{
+					ns:         m.Ns,
+					coll:       m.Coll,
+					pvtkeyHash: m.PvtkeyHash,
+				},
+				value: &purgeMarkerVal{
+					blkNum: blkNum,
+					txNum:  m.TxNum,
+				},
```

Comment:
The RFC says the blkNum and txNum will be part of the PurgeMarker key. I agree it should be in the 'value' not in the 'key'. Could you update the RFC accordingly?

Reply:
Sorry for the lengthy reply below but since its more of a practice question so writing my opinion in detail.

Yes, there has been some changes during the implementation, as it would always be the case. I am not sure whether we should go back to RFC each time for these kind of changes or rather better have a separate design doc that covers the details of a component in full. IMO, RFCs are mostly feature specific and cuts across components, and are useful as a initial review tools. In practice, a more usable artifact would be component design doc (that too, perhaps detailed code comments). For instance, if someone is looking at some code and wants to understand the overall design, it will be hard for him to know even which RFC to go for.

Having said that, I am not against updating the RFC. I'll do that at the end. Also, will update such that we don't have to treat the RFC as a detailed description of code. For instance, relating back to earlier comment, whether to delete in the background v/s during commit, we should be able to change it without going back to RFC.